api/_types/_core.search.d.ts

```diff
@@ -473,6 +486,10 @@ export type SuggestFuzziness = {
 
 export type SuggestSort = 'frequency' | 'score'
 
+export type T = Record<string, any>
+
+export type TDocument = Record<string, any>
+
```

Comment:
Eventually will be worth handling this a generic type parameters, but that's likely not a quick/easy lift.